### ISSUE TITLE
contract number

### ISSUE THREAD

**Issue by jseppi** [18F/calc#1486]

**Comment by jseppi** [comment]
Just saw we got another submission formatted like `GS-12--1234F` (notice the double `--`, which I think is just a typo).

**Comment by jseppi** [comment]
Consistent formatting and/or normalization is important also for how we determine if a price list has already been uploaded (at which point we show a message directing the user to the existing price list details page/modification flow).

**Comment by tram** [comment]
That's a great point! Any validation we do should probably let the CO know that it's important for that reason, too.

**Comment by jseppi** [comment]
@tram: could you do some sleuthing to figure out if there is a standard, predictable format?

**Comment by tram** [comment]
If the contract doesn't follow GS-XXX-XXXXX, we should at least throw a warning. We should also consider throwing a warning if the three characters after `GS-` don't match what we think it should be for that schedule.

From [Turbo GSA](http://www.turbogsa.com/gsa-faqs-copy):
> GSA contract numbers follow this format: GS-XXX-XXXXX.  The first two digits stand for General Schedule.  The next 3 digits refer to the specific schedule you have been awarded a contract under a specific GSA schedule. For example, an IT schedule 70 happens to use the code 35F. The final numbers (including a letter) are specific to your company.

**Comment by jseppi** [comment]
@ericronne: What about doing 2-text-box input type thing, kinda like we have for dates?

Rough sketch of what I mean:

<img width="265" alt="screen shot 2017-03-17 at 10 41 03 am" src="https://cloud.githubusercontent.com/assets/697848/24050878/463623aa-0afe-11e7-8b28-287dbc31b81b.png">

**Comment by ericronne** [comment]
I used that approach so we could implement it quickly; wanted to make it more elegant eventually. Could we use [something like fieldkit](http://square.github.io/field-kit/) to auto-format as they enter? I think @brendansudol is using this on login.gov.

![image](https://cloud.githubusercontent.com/assets/9259185/24051936/bf423ef2-0b01-11e7-9366-331aead676b9.png)

**Comment by jseppi** [comment]
Oh, you already did a design for this?

`fieldkit` looks very promising!

**Comment by ericronne** [comment]
No, i was referring to the original/current date-entry design. :)

**Comment by jseppi** [comment]
Oh ok -- this issue is about the contract number 😜

**Comment by ericronne** [comment]
I know. You referred to the date entry ;)

**Comment by ericronne** [comment]
Which i would love to redo in slicker fashion at some point.
I don't consider it a great paradigm.

**Comment by jseppi** [comment]
Ok, so your recommendation is to use some kind of auto-formatting library like `fieldkit` for the Contract Number input, right?

I think we can also do HTML5 input validation using the [`pattern` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-pattern) as a fallback for no-JS (or rather, use `fieldkit` as a progressive enhancement).

**Comment by ericronne** [comment]
Cool, and if it works well, perhaps we can use it on the date form, etc, at some point. Thx!

**Comment by toolness** [comment]
Note that the contract number (aka PIID) format has changed in the past year; see https://github.com/18F/calc/issues/1668#issuecomment-378561054 for more details.
